doc/page-properties.h

```diff
@@ -0,0 +1,129 @@
+/**********************************************************************
+**
+** Copyright (C) Luxoft Sweden AB 2020
+**
+** This file is part of the documentation of the FaceLift project
+**
+** This work is licensed under the Creative Commons
+** Attribution-ShareAlike 4.0 International License. To view a copy of
+** this license, visit http://creativecommons.org/licenses/by-sa/4.0/
+** or send a letter to Creative Commons, PO Box 1866, Mountain View,
+** CA 94042, USA.
+**
+** SPDX-License-Identifier: CC-BY-SA-4.0
+**
+**********************************************************************/
+
+/**
+\page page-properties Properties
+
+\section properties-intro Introduction
+Properties are the part of the interface definition described in *.qface file. Eventually they will be generated as C++ and QML properties of the interface. For more details about properties in *.qface files see documentation at https://github.com/Pelagicore/qface
+
+In Facelift each property is exposed to the QML engine as Q_PROPERTY with a setter and getter functions, and a change signal. If the property is marked as 'readonly' then no setter is generated.
+
+The facelift::Property class template provides encapsulation for the internal data with a const getter only (except for the service property), which guarantees that the data will not be unnoticibly changed. Each time the setter is called, with a value that differs from the previous one, the change signal is emitted. If the previous value equals to the new value, the change signal will not be emitted.
```

Comment:
You need to introduce the XXXImplementationBase class before talking about the facelift::Property template class.

Reply:
that line is too long

Reply:
Remove "If the previous value equals to the new value, the change signal will not be emitted.". The previous sentence is enough.

Reply:
"with a const getter only" ?? There are other methods in that class.
I doubt service properties have a specific here.

Reply:
> You need to introduce the XXXImplementationBase class before talking about the facelift::Property template class.

There will be separate page for it at some point. Reference will be added. Meanwhile I will add short information about it.

> that line is too long

This line is a whole paragraph in generated html, inside &lt;p&gt;&lt;/p&gt; html tags. Splitting this line would cause new paragraph generation.
> Remove "If the previous value equals to the new value, the change signal will not be emitted.". The previous sentence is enough.

OK

> "with a const getter only" ?? There are other methods in that class.

Service property returns const reference to a pointer. Sentence to be rephrased.